src/open-hackathon-server/Kaiyuanshe.OpenHackathon.Server/Controllers/UserController.cs

```diff
@@ -65,5 +67,27 @@ public class UserController : HackathonControllerBase
             return Ok(userInfo);
         }
         #endregion
+
+        #region GetUploadToken
+        /// <summary>
+        /// Get file upload token
+        /// </summary>
+        /// <returns>the SAS token</returns>
+        /// <response code="200">Success. The response return a SAS token.</response>
+        [HttpPost]
+        [ProducesResponseType(typeof(SasToken), StatusCodes.Status200OK)]
+        [SwaggerErrorResponse(400)]
+        [Route("user/generateSasToken")]
+        [Authorize(Policy = AuthConstant.PolicyForSwagger.LoginUser)]
+        public async Task<object> GetUploadToken([FromBody] FileUpload parameter)
+        {
+            var filename = parameter.filename;
+            var folder = CurrentUserId;
+            var token = UserManagement.GetUploadToken(parameter.expiration.Value, folder + "/" + filename);

```

Comment:
`int?` is nullable, you'd better use `expiration.GetVaultOrDefault()` to avoid NullException